build.py

```diff
@@ -740,7 +754,33 @@ def dockerfile_prepare_container_linux(argmap, backends):
          libre2-5 && \
     rm -rf /var/lib/apt/lists/*
 '''
-    df += install_dcgm_libraries(argmap['DCGM_VERSION'])
+    if enable_gpu:
+        df += install_dcgm_libraries(argmap['DCGM_VERSION'])
+        df += '''
+# Extra defensive wiring for CUDA Compat lib
+RUN ln -sf ${_CUDA_COMPAT_PATH}/lib.real ${_CUDA_COMPAT_PATH}/lib \
+ && echo ${_CUDA_COMPAT_PATH}/lib > /etc/ld.so.conf.d/00-cuda-compat.conf \
+ && ldconfig \
+ && rm -f ${_CUDA_COMPAT_PATH}/lib 
+'''
+    else:
+        df += '''
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+        git \
+        dirmngr \ 
+        libnuma-dev \
+        curl && \
+    rm -rf /var/lib/apt/lists/*
+'''
+        if 'onnxruntime' in backends:
+            df += '''
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+        libgomp1 && \
+    rm -rf /var/lib/apt/lists
```

Comment:
These are dependencies required for CPU only build? I wonder why we don't need them before (I couldn't find their reference in other place of `build.py`)

Reply:
It's because the base image is ubuntu:20.04 instead of cuda. I think the Nvidia cuda container installs OpenMP

Reply:
I think these deps we install in ubuntu:20.04 instead of our current base container should be handled elsewhere. We would want a single function that installs these additional deps and ensure that we do NOT try to install them again unless the base container is not our current tritonserver base container.